app/models/services/crossref.rb

```diff
@@ -0,0 +1,24 @@
+class Crossref
+  def initialize(options={})
+    @base_url = 'http://api.crossref.org/'
+    @metadata_url = @base_url + "works/"
+    @funder_url = @base_url + "funders/"
+  end
+
+  def get_doi_metadata(doi)
+    metadata_uri = URI.parse(@metadata_url+doi)
+    JSON.parse Net::HTTP.get(metadata_uri) if valid_json? Net::HTTP.get(metadata_uri)
+  end
+
+  private
+
+    def valid_json?(json)
+      begin
+        JSON.parse(json)
+        return true
+      rescue JSON::ParserError => e
+        return false
+      end
+    end
```

Comment:
Normally I'm not much for focusing on efficiency, but you are parsing the JSON 3 times. Also think about what the return value for the method should be. Right now if the condition fails, what's the return value? Whatever called the method will get some other kind of return value that may break things down the line.